sip/sip-004-materialized-view.md

```diff
@@ -0,0 +1,822 @@
+# SIP 004 Cryptographic Committment to Materialized Views
+
+## Preamble
+
+Title: Cryptograhpic Commitment to Materialized Views
+
+Author: Jude Nelson <jude@blockstack.com>
+
+Status: Draft
+
+Type: Standard
+
+Created: 7/15/2019
+
+License: BSD 2-Clause
+
+## Abstract
+
+Blockchain peers are replicated state machines, and as such, must maintain a
+materialized view of all of the state the transaction log represents in order to
+validate a subsequent transaction.  The Stacks blockchain in particular not only
+maintains a materialized view of the state of every fork, but also requires
+miners to cryptographically commit to that view whenever they mine a block.
+This document describes a **Merklized Adaptive Radix Trie** (MARF), an
```

Comment:
Why are we using MARF as an acronym of Merklized Adaptive Radix Trie? Why not MART?

Reply:
Should be "Merklized Adaptive Radix Forest".  Fixed!